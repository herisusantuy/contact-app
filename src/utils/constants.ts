export const testID = {
  swipeableWrapper: "card-contact-swipeable-wrapper",
  photoProfile: "card-contact-photo-profile",
  fullnameText: "card-contact-fullname-text",
  ageText: "card-contact-age-text",
  pressableView: "floating-button-pressable-view",
  plusIcon: "floating-button-plus-icon",
  textInput: "form-text-field",
  customModal: "custom-modal",
  cancelButton: "cancel-button",
  saveButton: "save-button",
};
