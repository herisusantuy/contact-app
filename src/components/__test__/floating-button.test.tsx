import { describe, expect, test } from "@jest/globals";
import { fireEvent } from "@testing-library/react-native";
import FloatingButton from "../floating-button";
import { renderWithProviders } from "../../utils/test-utils";
import { testID } from "../../utils/constants";

describe("Floating Button", () => {
  it("Should render Pressable view", () => {
    const { getByTestId } = renderWithProviders(
      <FloatingButton onPress={() => console.log("Button press!")} />
    );
    const pressableView = getByTestId(testID.pressableView);

    expect(pressableView).toBeTruthy();
  });
  it("Button call onPress function when it is pressed.", () => {
    const mockOnPress = jest.fn();
    const { getByTestId } = renderWithProviders(
      <FloatingButton onPress={mockOnPress} />
    );
    const pressableView = getByTestId(testID.pressableView);
    fireEvent.press(pressableView);
    expect(mockOnPress).toHaveBeenCalled();
  });

  it("Floating button render correctly..", () => {
    const mockOnPress = jest.fn();
    const tree = renderWithProviders(
      <FloatingButton onPress={mockOnPress} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
