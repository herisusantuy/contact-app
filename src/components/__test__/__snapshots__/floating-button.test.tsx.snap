// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Floating Button Floating button render correctly.. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "green",
      "borderColor": "green",
      "borderRadius": 100,
      "borderWidth": 1,
      "bottom": 100,
      "height": 70,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "right": 30,
      "width": 70,
    }
  }
  testID="floating-button-pressable-view"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "white",
          "fontSize": 32,
        },
        undefined,
        {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    testID="floating-button-plus-icon"
  >
    î˜§
  </Text>
</View>
`;
